packages/legend-query-builder/src/components/result/QueryBuilderResultPanel.tsx

```diff
@@ -638,12 +692,15 @@ export const QueryBuilderResultPanel = observer(
           <CubesLoadingIndicator isLoading={isLoading}>
             <CubesLoadingIndicatorIcon />
           </CubesLoadingIndicator>
-          {!executionResult && !isLoading && (
+          {!executionResult && !isLoading && !resultState.executionError && (
             <BlankPanelContent>
               Build or load a valid query first
             </BlankPanelContent>
           )}
-          {executionResult && !isLoading && (
+          {!isLoading && (
```

Comment:
does this need a !executionResult as well in the boolean condition?

Reply:
executionResult is not reset to be undefined if the second execution failed but the first execution succeeded.
Therefore adding `!executionResult` to the condition will cause error messages not to be displayed.
The new three conditions for showing `blank`, `error message`, `grid data` should have worked well as expected